### PR TITLE
BUG: .drop(fields). The argument can now be either a list of strings …

### LINKED ISSUE #2829: Weird behavior of .drop(fields)
```
import string

my_project = 'pj-becfr-eagle-ci-dev'
my_dataset = 'dm_express_assortment_V1'

import ibis
import ibis_bigquery
from google.cloud import bigquery
import pandas as pd

my_table = 'issue_3_table'

pdf = pd.DataFrame({'a': [1], 'b': [2], 'c': [3]})

# Load client
client = bigquery.Client(project=my_project)

# Load data to BQ
client.delete_table(my_dataset + "." + my_table, not_found_ok=True)
job = client.load_table_from_dataframe(pdf, my_dataset + "." + my_table)

ibis.options.interactive = False

conn = ibis_bigquery.connect(
    project_id=my_project,
    dataset_id=my_dataset)

t = conn.table(my_table)

e = t.drop("ab")

e.schema()  # a and b are removed

e = t.mutate(ab=t.c)

e.drop("ab")  # a and b are removed


e = t.mutate(ab=t.c)

e.drop(["ab"])  # ab is removed

e = t.mutate(bz=t.c)

e.drop("bz")  => KeyError: "Fields not in table: frozenset({'z'})"

e.drop(["bz"])  # bz is removed

e.drop(e.a) => ValueError: The truth value of an Ibis expression is not defined

e.drop([e.a]) => KeyError: "Fields not in table: frozenset({ref_0\nBigQueryTable[table]\n  name: ...
```

I think there is a mismatch between a string and a list of strings

The .drop() method takes a list/sequence/... of fields as argument.

From the examples above, each field can be a string (at least).

So t.drop(["ab", "bc"]) attempts to remove the attributes: "ab" and "bc" and that's fine.

It seems that if I call t.drop("ab"), ibis considers that the 2 attributes are "a" and "b". 
But, that's not correct because the elements of a string are characters and not strings, unless we 
suppose that field names can also be characters...

Anyway, that's quite  confusing.

### PR DESCRIPTION
**Pull request by jcmincke**
Closes #2829

### CODE REVIEW

File: ibis/expr/api.py
Comment (datapythonista) [inline ibis/expr/api.py:4425]: I'd prefer to use `fields = [fields]`, instead of recursively calling the function (which IMHO makes things more difficult to understand, and adds an overheat for no reason.

Also, maybe just personal preference, but seems like it'd make sense to keep the `if not fields:` first, and just check this `if` if fields is present.

File: docs/source/release/index.rst
Comment (datapythonista) [inline docs/source/release/index.rst:16]: If you check below, looks like for multi-line comments, we indent at the beginning of the test, not the bullet point (in short, can you indent this line two more spaces please).

File: ibis/tests/sql/test_compiler.py
Comment (datapythonista) [inline ibis/tests/sql/test_compiler.py:2574]:
```suggestion
    assert e1.schema() != t.schema()
```

File: ibis/tests/sql/test_compiler.py
Comment (datapythonista) [inline ibis/tests/sql/test_compiler.py:2576]:
```suggestion
    assert "b" not in e1.columns
```

**Review by datapythonista — COMMENTED** [review]
Thanks for the PR @jcmincke, looks good. Added few minor comments, but other than that, nice improvement!

File: docs/source/release/index.rst
Comment (datapythonista) [inline docs/source/release/index.rst:16]: These notes are for Ibis users. I think they'll care more on how this change affects them, `.drop()` now accepts single columns, than how we implemented this internally (transforming the string to on element list). Do you mind rephrasing a bit please? Sorry I didn't see this in the previous review. Thanks!

**Review by datapythonista — APPROVED** [review]
Thanks @jcmincke. Will wait to have a second pair of eyes here, but lgtm.

File: ibis/expr/api.py
Comment (jreback) [inline ibis/expr/api.py:4427]: hmm there isn't any doc-string, can you add one, also  type fields.

File: ibis/tests/sql/test_compiler.py
Comment (jreback) [inline ibis/tests/sql/test_compiler.py:2563]: can you add the issue number as a comment

File: ibis/tests/sql/test_compiler.py
Comment (jreback) [inline ibis/tests/sql/test_compiler.py:2567]: can you call this expected

File: ibis/tests/sql/test_compiler.py
Comment (jreback) [inline ibis/tests/sql/test_compiler.py:2568]: can you call this result1 and result2 for the one below

**Review by jreback — CHANGES_REQUESTED** [review]

File: ibis/expr/api.py
Comment (jreback) [inline ibis/expr/api.py:4427]: you can remove these 2 lines, they are extraneous

**Review by jreback — CHANGES_REQUESTED** [review]
small comment, ping on green

File: ibis/expr/api.py
Comment (datapythonista) [inline ibis/expr/api.py:4421]:
```suggestion
def _table_drop(self,
```
The convention with mypy is to not annotate `self`.

File: ibis/expr/api.py
Comment (jcmincke) [reply ibis/expr/api.py:4427]: I don't understand. You asked me: "hmm there isn't any doc-string, can you add one, also type fields."

Where should the doc-string be added?

File: ibis/expr/api.py
Comment (jcmincke) [reply ibis/expr/api.py:4421]: Do you also type the return value.

Comment (datapythonista) [reply ibis/expr/api.py:4421]: Yes. The idea is that mypy already knows which is the type of self, since it's the Python interpreter who provides it. But for the return, is the user (author of the function) who decides what is being returned, so mypy needs to be told.

File: ibis/expr/api.py
Comment (datapythonista) [reply ibis/expr/api.py:4427]: Docstring is fine where it is. Jeff's comment is that you can remove those two lines about the two calls being equivalent.

**Comment by jreback** [comment]
pls also merge master

**Comment by datapythonista** [comment]
@jcmincke do you have time to finish this? Just removing an annotation and a paragraph of the docstring is missing to merge this. Also, there is a small conflict in the release notes, if you can merge master and fix please. Thanks!

**Comment by datapythonista** [comment]
Thanks for the update @jcmincke, looks like you'll have to run black to get the CI green.

**Comment by datapythonista** [comment]
@jreback your comments were addressed, if you want to have a last look before merging...

**Review by jreback — APPROVED** [review]

**Comment by datapythonista** [comment]
Thanks for the contribution @jcmincke

**Pull request merged**
